cfme/tests/cli/test_appliance_console.py

```diff
@@ -1291,7 +1290,10 @@ def test_appliance_console_extend_storage_negative():
             2.
             3. Check there is no extra disk present.
     """
-    pass
+    command_set = ("ap", RETURN, "11")
+    result = appliance.appliance_console.run_commands(command_set, timeout=30, output=True)
+    assert [i for i in result[-1].split("\n") if "/dev/vd" not in i.strip()], (
```

Comment:
What is `result[-1]`?  `appliance_console.run_commands` is returning stdout - so isn't this referencing the last character in the stdout string ( a newline) and then trying to split on it?

This is a false positive, and will never fail.  You need to test your logic.

```In [1]: result = ''' 
   ...: test1 
   ...: test2 
   ...: test3 
   ...: '''                                                                                   

In [2]: [i for i in result[-1].split('\n') if '/dev/vd' not in i.strip()]                     
Out[2]: ['', '']

In [3]: assert [i for i in result[-1].split('\n') if '/dev/vd' not in i.strip()]              

```

Reply:
I might be wrong here though - add_commands might be a list of the stdout from each step in the commands. I think this still needs comments, as @john-dupuy suggested, about the assertion being made.

There's no check that /dev/vc is the last disk before the command is run, so how stable will this test be against multiple appliance types, running on different providers? The assertion needs to be stronger, and I think should be split up to be more readable. Name the part of stdout from run_commands that you want to assert against, and compare it to an original list of disks before the command was run.  That way the test isn't sensitive to the disk configuration of the particular appliance that its running against.

Reply:
I used `"/dev/vd"` for vda, vdb and so on but not getting you about `"dev/vc"`, could you please elaborate?